### PR TITLE
Adminfacility add remove

### PR DESCRIPTION
**Pull request by dominikl**
# What this PR does

Adds 'add' and 'remove' admin privileges to the Java Gateway.

# Testing this PR

Check that the `testAdminPrivileges` integration test passes.

# Related reading

https://trello.com/c/FjayBnvM/39-gateway-support-for-roles

### CODE REVIEW

File: components/blitz/src/omero/gateway/facility/AdminFacility.java [hunk 1]
```diff
@@ -370,6 +370,64 @@ public void setAdminPrivileges(SecurityContext ctx, ExperimenterData user,
...
+     * @throws DSAccessException
+     *             If an error occurred while trying to retrieve data from OMERO
+     *             service.
+     */
+    public void removeAdminPrivileges(SecurityContext ctx,
+            ExperimenterData user, List<String> privileges)
+            throws DSOutOfServiceException, DSAccessException {
+        try {
+            List<String> privs = getAdminPrivileges(ctx, user);
+            for (String priv : privileges)
```
Comment (mtbc) [inline components/blitz/src/omero/gateway/facility/AdminFacility.java]: just use `List.removeAll`?

File: components/blitz/src/omero/gateway/facility/AdminFacility.java [hunk 2]
```diff
@@ -370,6 +370,62 @@ public void setAdminPrivileges(SecurityContext ctx, ExperimenterData user,
...
+     * @param privileges
+     *            The admin privileges to grant.
+     * @throws DSOutOfServiceException
+     *             If the connection is broken, or not logged in.
+     * @throws DSAccessException
+     *             If an error occurred while trying to retrieve data from OMERO
+     *             service.
+     */
+    public void addAdminPrivileges(SecurityContext ctx, ExperimenterData user,
+            List<String> privileges) throws DSOutOfServiceException,
```
Comment (mtbc) [inline components/blitz/src/omero/gateway/facility/AdminFacility.java]: Can make `privileges` a `Collection<String>` without further changes to method?

File: components/blitz/src/omero/gateway/facility/AdminFacility.java [hunk 3]
```diff
@@ -370,6 +370,62 @@ public void setAdminPrivileges(SecurityContext ctx, ExperimenterData user,
...
+     * @param privileges
+     *            The admin privileges to revoke.
+     * @throws DSOutOfServiceException
+     *             If the connection is broken, or not logged in.
+     * @throws DSAccessException
+     *             If an error occurred while trying to retrieve data from OMERO
+     *             service.
+     */
+    public void removeAdminPrivileges(SecurityContext ctx,
+            ExperimenterData user, List<String> privileges)
```
Comment (mtbc) [inline components/blitz/src/omero/gateway/facility/AdminFacility.java]: Can make `privileges` a `Collection<String>` without further changes to method?

**Comment by mtbc** [comment]
https://ci.openmicroscopy.org/job/OMERO-DEV-merge-integration-java/lastCompletedBuild/testngreports/integration.gateway/AdminFacilityTest/ is green.

**Comment by mtbc** [comment]
Looking good overall, just offered a small suggestion.

**Comment by dominikl** [comment]
Thanks @mtbc , changed it to generally use `Collection` instead of `List`.

**Review by mtbc — APPROVED** [review]

**Pull request merged**